packages/tab-scroller/index.tsx

```diff
@@ -0,0 +1,276 @@
+      {[styleName]: value}
+    );
+    this.setState((prevState) => {
+      return Object.assign(prevState, {
+        [elementStyleProperty]: updateElementStyleProperty,
+      });
+    });
+  };
+
+  get adapter() {
+    return {
+      eventTargetMatchesSelector: (evtTarget: HTMLDivElement, selector: string) => {
+        if (selector && isElement(selector)) {
+          return evtTarget[MATCHES as 'matches'](selector);
+        }
+        return false;
+      },
+      addClass: (className: string) => {
+        const classList = new Set(this.state.classList);
+        classList.add(className);
+        this.setState({classList});
+      },
+      removeClass: (className: string) => {
+        const classList = new Set(this.state.classList);
+        classList.delete(className);
+        this.setState({classList});
+      },
+      addScrollAreaClass: (className: string) => {
+        const areaClassList = new Set(this.state.areaClassList);
+        areaClassList.add(className);
+        this.setState({areaClassList});
+      },
+      setScrollAreaStyleProperty: (prop: string, value: string) =>
+        this.setStyleToElement(prop, value, 'scrollAreaStyleProperty'),
+      setScrollContentStyleProperty: (prop: string, value: string) =>
+        this.setStyleToElement(prop, value, 'scrollContentStyleProperty'),
+      getScrollContentStyleValue: (propName: string) =>
+        this.contentElement_.current &&
+        window
+          .getComputedStyle(this.contentElement_.current)
+          .getPropertyValue(propName),
+      setScrollAreaScrollLeft: (scrollX: number) => {
+        if (!this.areaElement_.current) return;
+        this.areaElement_.current.scrollLeft = scrollX;
+      },
+      getScrollAreaScrollLeft: () =>
+        this.areaElement_.current && this.areaElement_.current.scrollLeft,
+      getScrollContentOffsetWidth: this.getScrollContentWidth,
+      getScrollAreaOffsetWidth: () =>
+        this.areaElement_.current && this.areaElement_.current.offsetWidth,
+      computeScrollAreaClientRect: () =>
+        this.areaElement_.current &&
+        this.areaElement_.current.getBoundingClientRect(),
+      computeScrollContentClientRect: () =>
+        this.contentElement_.current &&
+        this.contentElement_.current.getBoundingClientRect(),
+      computeHorizontalScrollbarHeight: () =>
+        util.computeHorizontalScrollbarHeight(document),
+    };
+  }
+
+  getScrollPosition = () => {
+    return this.foundation_.getScrollPosition();
+  };
+
+  // needs to be public class method for react tab-bar
+  getScrollContentWidth = () => {
+    return (
+      this.contentElement_.current && this.contentElement_.current.offsetWidth
+    );
+  };
+
+  incrementScroll = (scrollXIncrement: number) => {
+    this.foundation_.incrementScroll(scrollXIncrement);
+  };
+
+  scrollTo = (scrollX: number) => {
+    this.foundation_.scrollTo(scrollX);
+  };
+
+  handleWheel_ = (evt: React.WheelEvent<HTMLDivElement>) => {
```

Comment:
Can this and the other underscore-suffixed instance field functions be private?

Reply:
all 3 of these 
```
areaElement_: React.RefObject<HTMLDivElement> = React.createRef();
  contentElement_: React.RefObject<HTMLDivElement> = React.createRef();
  foundation_?: MDCTabScrollerFoundation;```
are referenced in the tests. 

I did make the foundation non-nullable though.